### PR TITLE
Feature/update readme

### PR DESCRIPTION
**Pull request by formatc1702**
I've restructured the [main README file](https://github.com/formatc1702/WireViz/blob/feature/update-readme/README.md), added and moved around some of the information. I welcome any new input to make the readme more user-friendly.

### CODE REVIEW

**Comment by formatc1702** [comment]
A better description for the `build_examples.py` options is probably a good thing to add.

File: README.md [hunk 1]
```diff
@@ -90,8 +81,39 @@ Output file:
...
+
+#### Installing the development version
+
+Access to the current state of the development branch can be gained by cloning the repo and installing manually:
+
+```
+git clone <repo url>
+cd <working copy>
+git checkout dev
+pip3 install -e .
```
Comment (formatc1702) [inline README.md:112]: Please make sure this workflow makes sense.

**Comment by kvid** [comment]
> A better description for the `build_examples.py` options is probably a good thing to add.

See PR #118 description for a short summary in the Usage section, and a couple of use cases in the last section, to get some ideas.

**Comment by formatc1702** [comment]
It might make sense to create a `docs/` directory or similar, for the changelog, the syntax description, and the `build_examples.py` documentation. This way we don't overload the readme, or the parent directory.
In the future, this can be replaced by a properly built documentation system.

File: README.md (hunk 1 above)
Comment (kvid) [reply README.md:112]: Should you also mention something like the alternative variant you gave to me in https://github.com/formatc1702/WireViz/issues/43#issuecomment-651078861?

```sh
git clone https://github.com/formatc1702/WireViz.git
cd WireViz/
python -m venv wv-env
source wv-env/bin/activate
pip3 install -e .
cd src/wireviz
python build_examples.py
```

Note that `git checkout dev` is missing above, and I have not tried any of these installations from scratch recently.

Comment (kvid) [reply README.md:112]: Should there be a reference to CONTRIBUTING.md in the section about Installing the development version?

**Comment by formatc1702** [comment]
I'm merging this into `feature/syntax-description` since they are closely related; let's continue the discussion there.

File: README.md (hunk 1 above)
Comment (formatc1702) [reply README.md:112]: Sorry, didn't see these comments earlier.
- I'm not sure whether we should give advice on virtual environments directly, or reference a site that explains the concept more generally.
- Sure, I can include a link to `CONTRIBUTING.md`

**Pull request merged**